### PR TITLE
OPS-1449 Refactor agreement/{id}/history endpoint

### PR DESCRIPTION
**Pull request by rajohnson90**
## What changed

This is one small part of the tech debt item OPS-1449 where we are trying to restrict the depth our rest API goes to when nesting routes. This PR fixes one route, the agreements/{id}/history endpoint to be agreement-history/{id} so that the endpoint matches our standard defined in the tech debt.

## Issue

[OPS-1449](https://github.com/HHS/OPRE-OPS/issues/1449)

## How to test
1. Run Python unit tests
2. Log in as basic user, then open any agreement. Request to move a budget line to planned or in execution.
3. Log in as division director, then approve that agreement.
4. Open the agreement and look at the agreement history. You should still see at least the 2 items corresponding to this interaction in the agreement history page.

## Screenshots

image
A screenshot of the agreement history  on localhost after my refactor. If you are testing this change, you should also see something similar. 

## Definition of Done Checklist
- [x] OESA: Code refactored for clarity
- [x] OESA: Dependency rules followed
- [x] Automated unit tests updated and passed
- [x] Automated integration tests updated and passed
- [x] Automated quality tests updated and passed
- [x] Automated load tests updated and passed
- [x] Automated a11y tests updated and passed
- [x] Automated security tests updated and passed
- [x] 90%+ Code coverage achieved
- [x] Form validations updated

### CODE REVIEW

**Comment by jonnalley** [comment]
👏

**Comment by maiyerlee** [comment]
Looks great! Should we update the endpoint documentation in the openapi.yml file too? I see that line 167 is still `/api/v1/agreements/{agreement_id}/history`

File: frontend/src/api/getAgreementHistory.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ export const getAgreementHistoryByIdAndPage = async (id, page) => {
     const limit = pageSize;
     const offset = pageSize * (page - 1);
     const api_version = ApplicationContext.get().helpers().backEndConfig.apiVersion;
-    const endpoint = `/api/${api_version}/agreements/${id}/history/?limit=${limit}&offset=${offset}`;
+    const endpoint = `/api/${api_version}/agreement-history/${id}?limit=${limit}&offset=${offset}`;
```
Comment (fpigeonjr) [inline frontend/src/api/getAgreementHistory.js:8]: 🟨 reminder to add tech-debt to use RTK Query in favor of this approach

**Review by fpigeonjr — APPROVED** [review]
💪 tested locally and works great

**Comment by rajohnson90** [comment]
Maiyer is right, I should definitely update the openapi.yml file. I will get that done before merging.

**Pull request merged**